### PR TITLE
update eslint & typescript configs & other chores

### PR DESCRIPTION
**Pull request by okvv**
* remove `plugin:@typescript-eslint/recommended-requiring-type-checking` as it's too restricting for a sample code
* fix global module type definitions
* add ts-node, which won't work without the fix
* add package json with npm run scripts
* update readme with new npm run scripts
* fix redundant ts errors, cleanup eslint log
* globalThis not backward compatible, revert to global
* bug fix: typo in license type Licence instead of License
* add node 10 bundle configs with polyfill for import.url
* add npm script to produce it
* rename restart command to reboot
* re-bundle updated code
* update tests after config file path changes

### CODE REVIEW

**Comment by borisd** [comment]
Task linked: [CU-2p6k5x1 update eslint typescript config](https://app.clickup.com/t/2p6k5x1)

**Review by morsdyce — DISMISSED** [review]

File: src/main.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import authenticateDevice from './authentication.js';
 import notifyServerLoop from './scheduler.js';
 import { INITIAL_APP_STATE, INTERVAL_IN_MS } from './helpers/constants.js';
 
-global.applicationState = INITIAL_APP_STATE;
+globalThis.applicationState = INITIAL_APP_STATE;
```
Comment (morsdyce) [inline src/main.ts:5]: there's a chance this might break in node 10

Comment (okvv) [reply src/main.ts:5]: you are correct. esbuild does not handle this case.

File: src/helpers/storage.ts [hunk 2]
```diff
@@ -1,31 +1,38 @@
 import fs from 'fs';
 import path from 'path';
+import url from 'url';
 import { CONFIG_FILE_NAME, ERR_LOG_FILE, FIRMWARE_FILE_NAME, INITIAL_APP_STATE, STD_LOG_FILE } from './constants.js';
 import { State } from './types';
 
+const dirname = Boolean(import.meta.url) ? url.fileURLToPath(new URL('.', import.meta.url)) : __dirname;
```
Comment (morsdyce) [inline src/helpers/storage.ts:7]: Did you test this in node 10? it could fail on import being undefined

Comment (okvv) [reply src/helpers/storage.ts:7]: yes. tested it in all 4 ways of running this project

**Review by morsdyce — APPROVED** [review]

**Pull request merged**